modules/ximgproc/src/paillou_filter.cpp

```diff
@@ -0,0 +1,454 @@
+#include "precomp.hpp"
+#include "opencv2/highgui.hpp"
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+namespace cv {
+namespace ximgproc {
+
+/*
+If you use this code please cite this @cite paillou1997detecting
+Detecting step edges in noisy SAR images: a new linear operator  IEEE Transactions on Geoscience and Remote Sensing  (Volume:35 ,  Issue: 1 )  1997
+*/
+
+
+class ParallelGradientPaillouYCols: public ParallelLoopBody
+{
+private:
+    Mat &img;
+    Mat &im1;
+    double a;
+    double w;
+    bool verbose;
+public:
+    ParallelGradientPaillouYCols(Mat& imgSrc, Mat &d,double aa,double ww):
+        img(imgSrc),
+        im1(d),
+        a(aa),
+        w(ww),
+        verbose(false)
+    {}
+    void Verbose(bool b){verbose=b;}
+    virtual void operator()(const Range& range) const
+    {
+        if (verbose)
+            std::cout << getThreadNum()<<"# :Start from row " << range.start << " to "  << range.end-1<<" ("<<range.end-range.start<<" loops)" << std::endl;
+
+        float                *f2;
+        int tailleSequence=(img.rows>img.cols)?img.rows:img.cols;
+        double *yp=new double[tailleSequence],*ym=new double[tailleSequence];
```

Comment:
please consider using opencv classes for storing this data, for example cv::Vec, cv::Mat. to my mind it is better not to use raw pointers